### PR TITLE
Hello director

### PR DESCRIPTION
**Pull request by odeimaiz**
- Results Folder added. Displays the directory structure in a tree view
- Results Viewer added. Displays:

1. - Images
2. - HTMLs
3. - CSVs

- Refactoring server and client sides
- Cleaning code

### CODE REVIEW

File: demos/frontend-react/app/src/reducers/workbenchServiceReducer.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ const newServiceReducer = function(state = initialState.workbench, action) {
     let copiedService = copyObject(action.payload);
     let newNode = {
       uniqueName: copiedService.text + '_S' + Number(newState.nodes.length+1),
-      service: copiedService
+      service: copiedService,
```
Comment (oetiker) [inline demos/frontend-react/app/src/reducers/workbenchServiceReducer.js]: a dangling comma ?

File: demos/frontend-react/app/src/components/ResultsViewer.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+          }}
+          visibility = {this.state.visible}
+          size = {{ width: this.state.width,  height: this.state.height }}
+          position = {{ x: this.state.x, y: this.state.y }}
+          onDragStop = {(e, d) => { this.setState({ x: d.x, y: d.y }) }}
+          onResize = {(e, direction, ref, delta, position) => {
+            this.setState({
+              width: ref.offsetWidth,
+              height: ref.offsetHeight,
+              ...position,
```
Comment (oetiker) [inline demos/frontend-react/app/src/components/ResultsViewer.js]: a dangling comma ?

Comment (odeimaiz) [reply demos/frontend-react/app/src/components/ResultsViewer.js]: I must admit it was a 'mistake', but according to the Airbnb JavaScript Style Guide, it is correct:
https://github.com/airbnb/javascript#commas

I should be consistent through the entire code though!

Comment (oetiker) [reply demos/frontend-react/app/src/components/ResultsViewer.js]: well it just means you rely on a transpilation step ... from a coding point of view the dangling comma is certainly preferable. most frameworks transpile with babel anyway theses days, so I guess you are fine

**Comment by oetiker** [comment]
it might make sense adding a linting step via travis so that incoming PRs get checked for compliance

**Pull request merged**